torch/distributed/checkpoint/state_dict.py

```diff
@@ -77,9 +82,13 @@ class StateDictOptions:
     """
     This dataclass specifies how get_state_dict/set_state_dict will work.
 
-    - ``fsdp_state_dict_type``: if the model contains FSDP sharded submodules,
-      what FSDP state_dict type should be used.
-      The default value is SHARDED_STATE_DICT.
+    - ``full_state_dict``: if this is set to True, all the tensors in the
```

Comment:
@fegin n00b question - Assuming `AsyncCollectiveTensors` are still returned without waiting?

Reply:
I'll update the dtensor gathering to use the `full_tensor()` API in a follow-up PR, which will always return synchronously so we can make sure that the value in the state_dict is correct.